.circleci/local_publish_helpers.sh

```diff
@@ -64,6 +64,9 @@ function generatePkgCli {
   # Restore .d.ts files required by @aws-amplify/codegen-ui at runtime
   cp ../node_modules/typescript/lib/*.d.ts node_modules/typescript/lib/
 
+  # Set NODE_ENV to production
+  sed -i "2s/^/process.env.NODE_ENV = 'production';\n/" node_modules/@aws-amplify/cli-internal/bin/amplify
+
```

Comment:
I agree - I think this would be a good opportunity to create an either build time or runtime injection mechanism so that there is dev/prod parity. We can either configure the build environments with the correct env vars or inject the Node process with environment variables at runtime.

There is the `dotenv` package: https://www.npmjs.com/package/dotenv that does something like this, but it is only supposed to be used for development so as not to encourage committing secrets. I like the idea of having non-private configurations near the code base though, and we inject the the environment variables based on the build target. Or we can configure the CI builds with the build targets env vars.